packages/synapse-react-client/src/components/SchemaDrivenAnnotationEditor/SchemaDrivenAnnotationsEditor.integration.test.tsx

```diff
@@ -348,7 +347,85 @@ describe('SchemaDrivenAnnotationEditor tests', () => {
       ...schemaHandlers,
       successfulUpdateHandler,
     )
-    await renderComponent()
+    const { container } = await renderComponent()
+    await screen.findByText('requires scientific annotations', { exact: false })
+
+    await waitFor(() => {
+      const countryInput = container.querySelector(
+        '#root_country',
+      ) as HTMLInputElement
+      expect(countryInput.value).toBe('USA')
+    })
+
+    // Saving the form should maintain the existing annotations
+    await clickSave()
+    await waitFor(() =>
+      expect(updatedJsonCaptor).toHaveBeenCalledWith(
+        expect.objectContaining({ country: 'USA', state: 'Washington' }),
+      ),
+    )
+  })
+
+  it('Reloads the form data properly after remounting the component (SWC-6486)', async () => {
+    server.use(
+      annotationsWithSchemaHandler,
+      ...schemaHandlers,
+      successfulUpdateHandler,
+    )
+    const { container, rerender } = await renderComponent()
+    const schemaAlert = await screen.findByText(
+      'requires scientific annotations',
+      { exact: false },
+    )
+
+    await waitFor(() => {
+      const countryInput = container.querySelector(
+        '#root_country',
+      ) as HTMLInputElement
+      expect(countryInput.value).toBe('USA')
+    })
+
+    // Clear the component to ensure we get a full re-render
+    await Promise.all([
+      waitForElementToBeRemoved(schemaAlert),
+      act(() => {
+        rerender(<></>)
+      }),
+    ])
+
+    // Re-rendering the editor should reload the existing annotations
+    act(() => {
+      rerender(<SchemaDrivenAnnotationEditor {...defaultProps} />)
+    })
+    await screen.findByText('requires scientific annotations', { exact: false })
+
+    await waitFor(() => {
+      const countryInput = container.querySelector(
+        '#root_country',
+      ) as HTMLInputElement
+      expect(countryInput.value).toBe('USA')
+    })
```

Comment:
Now the cache is hot. When the bug occurred, the input field never got the value